pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -671,10 +658,10 @@ def _save_last_checkpoint(self, trainer: "pl.Trainer", monitor_candidates: Dict[
         filepath = self._format_checkpoint_name(self.CHECKPOINT_NAME_LAST, monitor_candidates)
         filepath = os.path.join(self.dirpath, f"{filepath}{self.FILE_EXTENSION}")
 
-        self._save_model(trainer, filepath)
+        trainer.save_checkpoint(filepath, self.save_weights_only)
 
-        if self.last_model_path and self.last_model_path != filepath and trainer.should_rank_save_checkpoint:
```

Comment:
what happens with this trainer property now, will you remove it? the only use was for the callback here.

Reply:
@awaelchli the plan is to deprecate it: https://github.com/PyTorchLightning/pytorch-lightning/issues/9074

Reply:
Yup, that's the follow up PR! :)

Reply:
ah, didn't read carefully. great